usr/share/rear/lib/layout-functions.sh

```diff
@@ -819,6 +819,40 @@ is_disk_a_pv() {
     return 0
 }
 
+# Check whether disk is suitable for being added to layout
+# Can be used to skip obviously unsuitable/broken devices
+# (missing device node, zero size, device can't be opened).
+# Should not be used to skip potential mapping targets before layout restoration
+# - an invalid disk may become valid later, for example if it is a DASD that needs
+# low-level formatting (see 090_include_dasd_code.sh and 360_generate_dasd_format_code.sh),
+# unformatted DASDs show zero size.
+# Returns 0 if the device is ok
+# Returns nonzero code if it should be skipped, and a text describing the error
+# (without the device name) on stdout
+# usage example:
+# local err
+# if ! err=$(is_disk_valid /dev/sda); then
+
+function is_disk_valid {
+    local disk="$1"
+    local size
+
+    if ! test -b "$disk" ; then
+        echo "$disk is not a block device"
+        return 1
+    fi
+    # capture stdout in a variable and redirect stderr to stdout - the error message
+    # will be our output
+    if { size=$(blockdev --getsize64 "$disk") ; } 2>&1 ; then
+        if ! test "$size" -gt 0 2>/dev/null ; then
+            echo "$disk has invalid size $size"
+            return 1
+        fi
```

Comment:
For easier understanding of the logic
I am wondering if an explicit 'returm 0' here
would be good?
At least to me it looked as if is_disk_valid()
would always 'return 1'.

Reply:
Or with "Return early, return often" style, cf.
https://github.com/rear/rear/wiki/Coding-Style#return-early-return-often
perhaps even like:
```
    if { size=$( blockdev --getsize64 "$disk" ) ; } 2>&1 ; then
        test "$size" -gt 0 2>/dev/null && return 0
        echo "$disk has invalid size $size"
        return 1
    fi
    return 1
```

Reply:
> Or with "Return early, return often" style, cf. https://github.com/rear/rear/wiki/Coding-Style#return-early-return-often perhaps even like:
> 
> ```
>     if { size=$( blockdev --getsize64 "$disk" ) ; } 2>&1 ; then
>         test "$size" -gt 0 2>/dev/null && return 0
>         echo "$disk has invalid size $size"
>         return 1
>     fi
>     return 1
> ```

This also looks like the function always returns 1 - `return 0` at the end of a complex line of code is easy to miss.

Reply:
A side note:
I wished I could do "post-if" like
```
return 0 if test "$size" -gt 0 2>/dev/null
```
instead of
```
test "$size" -gt 0 2>/dev/null && return 0
```